.github/workflows/tests.yml

```diff
@@ -24,9 +24,9 @@ jobs:
     - name: Build
       run: cargo build --verbose
     - name: Meilisearch (latest version) setup with Docker
-      run: docker run -d -p 7700:7700 getmeili/meilisearch:latest ./meilisearch --no-analytics=true --master-key=masterKey
+      run: docker run -d -p 7700:7700 getmeili/meilisearch:latest ./meilisearch --no-analytics --master-key=masterKey
     - name: Run tests
-      run: cargo test --verbose -- --test-threads=1
+      run: cargo test --verbose
```

Comment:
Even with the updates you made to testing with the macro, very nice update btw, I think tests will still have to be run with `--test-threads=1`. Meilisearch seems to deadlock when running multiple threads even if the indexes are different. We saw the same [issue](https://github.com/meilisearch/meilisearch-dart/issues/57) in the Dart client even with unique indexes so it isn't specific to the Rust client.

Reply:
Huuuum thanks I was not aware of this issue, I did a lot of tests locally and never triggered this issue though, are we sure it’s still present?
I’ll rerun it a bunch of time in the CI just to be sure it was not just a one time lucky run 👍

Reply:
I have not tried testing with multiple thread on 0.25 so it's possible it is fixed. In the past the deadlocks were random, but frequent enough that if you have run it a bunch of times I think you would have seen it. Were you running the test locally also, or just GH Action? It was vary rare for it to happen on GH since less cores are available.

Hopefully it isn't an issue any more and the thread limit can be removed :smile:

Reply:
Ah yes!
I’m pretty sure it has been solved on the v0.25 since we rewrote entirely the scheduler and made the creation and deletion of indexes asynchronous (this was the cause of all our issues)

Reply:
Hello @sanders41 👋 
Yes I think it's solved with v0.25.0 :) If it's not, definitely an issue to open on the core side